### PR TITLE
Service Accounts: Read password

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 1]
```diff
@@ -1123,6 +1123,70 @@ private void sleep() {
...
+					put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
+					build()));
+			return ResponseEntity.status(HttpStatus.FORBIDDEN).body("{\"errors\":[\"Permission Denied. Unable to get password details for the given service account\"]}");
+
+		}
+		else {
+			log.error(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+					put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
+					put(LogMessage.ACTION, "readSvcAccPassword").
+					put(LogMessage.MESSAGE, String.format("Unable to read password for [%s]", svcAccName)).
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Service account is printed in logs in some places and not in some places. Need to be consistent. No password should be logges

File: tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/ServiceAccountsServiceTest.java [hunk 2]
```diff
@@ -1272,18 +1272,16 @@ public void test_resetSvcAccPassword_success() {
...
-
         ADServiceAccountCreds adServiceAccountCreds = new ADServiceAccountCreds();
         adServiceAccountCreds.setCurrent_password("current_password");
         adServiceAccountCreds.setLast_password("last_password");
         adServiceAccountCreds.setUsername(svcAccName);
         String expectedOutput = getJSON(adServiceAccountCreds);
         Response pwdReadResponse = getMockResponse(HttpStatus.OK, true,expectedOutput);
         when(JSONUtil.getJSON(Mockito.any(ADServiceAccountCreds.class))).thenReturn(expectedOutput);
+
+        when(reqProcessor.process(Mockito.eq("/ad/serviceaccount/resetpwd"),Mockito.anyString(),Mockito.eq(token))).thenReturn(pwdReadResponse);
```
Comment (sivanag1974) [inline tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/ServiceAccountsServiceTest.java]: Why is there inconsistency in read/reset.  Reset is returning read response

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Updated service account name is logs

File: tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/ServiceAccountsServiceTest.java (hunk 2 above)
Comment (sarathmohan4) [reply tvaultapi/src/test/java/com/tmobile/cso/vault/api/service/ServiceAccountsServiceTest.java]: updated

**Pull request merged**